runtime/src/main/java/org/corfudb/recovery/FastObjectLoader.java

```diff
@@ -248,7 +258,10 @@ private void findAndSetLogHead() {
     }
 
     private void findAndSetLogTail() {
-        logTail = runtime.getSequencerView().query().getTokenValue();
+        Layout newLayout = runtime.getLayoutView().getLayout();
+        Layout.LayoutSegment latestSegment = newLayout.getSegments()
+                .get(newLayout.getSegments().size() - 1);
+        logTail = getMaxGlobalTail(newLayout, latestSegment, runtime);
```

Comment:
Can we add a log statement here? Displaying both the log tail and the epoch at which was calculated.
And one for the log head calculated too. This can help us catch bugs like the hole-filling one.

Reply:
Done

Reply:
can getMaxGlobalTail throw a network exception ? where is this retried ?

Reply:
Yes it could, moved getMaxGlobalTail logic to AddressView so we could rely on the layouthelper to retry and run systemdownHandler in case we fail to ever connect and fetch the log tail...